src/main/java/org/cyclops/everlastingabilities/recipe/TotemRecycleRecipe.java

```diff
@@ -0,0 +1,108 @@
+package org.cyclops.everlastingabilities.recipe;
+
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.EnumRarity;
+import net.minecraft.item.Item;
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.world.World;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.common.ForgeHooks;
+import org.cyclops.everlastingabilities.api.capability.IAbilityStore;
+import org.cyclops.everlastingabilities.api.capability.IMutableAbilityStore;
+import org.cyclops.everlastingabilities.item.ItemAbilityTotem;
+import org.cyclops.everlastingabilities.item.ItemAbilityTotemConfig;
+import org.cyclops.everlastingabilities.ability.AbilityHelpers;
+import org.cyclops.everlastingabilities.capability.MutableAbilityStoreConfig;
+import java.util.Random;
+
+public class TotemRecycleRecipe implements IRecipe {
+
+    private World _world = null;
+    
+    @Override
+    public boolean matches(InventoryCrafting invCrafting, World world) {
+    
+        _world = world;
+
+        int inputCount = 0;
+        for (int i = 0; i < invCrafting.getSizeInventory(); i++) {
+            ItemStack slot = invCrafting.getStackInSlot(i);
+            if (!slot.isEmpty()) {
+                if (slot.getItem() instanceof ItemAbilityTotem) {
+                    inputCount++;
+                }
+                else {
+                    // non-totem item found in recipe
+                    return false;
+                }
+            }
+        }
+        return inputCount == ItemAbilityTotemConfig.totemCraftingCount;
+    }
+    
+    @Override
+    public ItemStack getCraftingResult(InventoryCrafting invCrafting) {
+        if (_world.isRemote) {
```

Comment:
CyclopsCore has a method `MinecraftHelpers.isClientSide()`, this would avoid having to store the world.
Storing the world has two downsides:

1. If an external mod calls `getCraftingResult` without first calling `matches`, the world might still be `null`.
2. This will cause worlds to leak if they are unloaded but still stored this recipe.